lib/cusparse/linalg.jl

```diff
@@ -25,12 +25,137 @@ function sum_dim1(A::CuSparseMatrixCSR)
     return rowsum
 end
 
+function sum_dim2(A::CuSparseMatrixCSR)
+    function kernel(Tnorm, out, dA)
+        idx = (blockIdx().x-1) * blockDim().x + threadIdx().x
+        idx < length(dA.colPtr) || return
+        s = zero(Tnorm)
+        for k in dA.colPtr[idx]:dA.colPtr[idx+1]-1
+            s += abs(dA.nzVal[k])
+        end
+        out[idx] = s
+        return
+    end
+
+    A = CuSparseMatrixCSC(A)
+    m, n = size(A)
+    Tnorm = typeof(float(real(zero(eltype(A)))))
```

Comment:
When is `typeof(zero(eltype(A)))` != `eltype(A)`?

Reply:
It should be fixed now.